controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -274,7 +280,22 @@ func (r *RateLimitPolicyReconciler) cleanUpOrphanWASMPlugins(ctx context.Context
 			if err != nil {
 				return err
 			}
-			err = r.finalizeSingleWASMPlugins(ctx, rlp, wasmplugin)
+			wasmPluginDeleted, err = r.finalizeSingleWASMPlugins(ctx, rlp, wasmplugin)
+			if err != nil {
+				return err
+			}
+		}
+
+		// finalize pre-requisite of WasmPlugin i.e. EnvoyFilter adding the limitador cluster entry
+		if wasmPluginDeleted {
+			ef := &istionetworkingv1alpha3.EnvoyFilter{}
+			efKey := client.ObjectKey{Namespace: gwKey.Namespace, Name: kuadrantistioutils.LimitadorClusterEnvoyFilterName}
+			err := r.Client().Get(ctx, efKey, ef)
```

Comment:
why not delete and skip isNotFound errors?

Reply:
Good find :)